operator/api/redpanda/v1alpha2/common.go

```diff
@@ -38,6 +42,12 @@ type KafkaAPISpec struct {
 }
 
 // KafkaSASL configures credentials to connect to Redpanda cluster that has authentication enabled.
+// +kubebuilder:validation:XValidation:message="username and passwordSecretRef must be set when mechanism is plain",rule="self.mechanism.lowerAscii() != 'plain' || (self.username != \"\" && has(self.passwordSecretRef))"
+// +kubebuilder:validation:XValidation:message="username and passwordSecretRef must be set when mechanism is sha-256",rule="self.mechanism.lowerAscii() != 'scram-sha-256' || (self.username != \"\" && has(self.passwordSecretRef))"
+// +kubebuilder:validation:XValidation:message="username and passwordSecretRef must be set when mechanism is sha-512",rule="self.mechanism.lowerAscii() != 'scram-sha-512' || (self.username != \"\" && has(self.passwordSecretRef))"
+// +kubebuilder:validation:XValidation:message="oauth must be set when mechanism is oauth",rule="self.mechanism.lowerAscii() != 'oauthbearer' || has(self.oauth)"
```

Comment:
Shouldn't these be `&&` instead of `||` (or xor if that's support)? As is, this would error if any authentication methods (in this case `Oauth`) have been set. 

Would you mind adding a few tests for these cases as well?